src/coreclr/vm/jitinterface.cpp

```diff
@@ -1721,6 +1719,40 @@ bool CEEInfo::isFieldStatic(CORINFO_FIELD_HANDLE fldHnd)
     return res;
 }
 
+int CEEInfo::getArrayOrStringLength(CORINFO_OBJECT_HANDLE objHnd)
+{
+    CONTRACTL {
+        THROWS;
+        GC_TRIGGERS;
+        MODE_PREEMPTIVE;
+    } CONTRACTL_END;
+
+    int arrLen = -1;
+    JIT_TO_EE_TRANSITION();
+
+    _ASSERT(objHnd != NULL);
+
+    GCX_COOP();
+
+    Object* obj = OBJECTREFToObject(getObjectFromJitHandle(objHnd));
+
+    if (obj->GetMethodTable()->IsArray())
+    {
+        arrLen = ((ArrayBase*)obj)->GetNumComponents();
+    }
+    else if (obj->GetMethodTable()->IsString())
+    {
+        arrLen = ((StringObject*)obj)->GetStringLength();
+    }
+    else
+    {
+        UNREACHABLE();
+    }
+
+    EE_TO_JIT_TRANSITION();
```

Comment:
```suggestion
    EE_TO_JIT_TRANSITION_LEAF();
```

Reply:
Scratch this - this cannot be leaf. I have missed that it is switching GC modes.